LCM/scripts/PerformInventory.py

```diff
@@ -99,8 +101,8 @@ def usage():
     sys.exit(1)
 
 # Acquire inventory file lock
-filehandle = os.open(dsc_sysconfdir + "/inventory_lock", os.O_WRONLY | os.O_CREAT , 0o644)
-inventory_lock = os.fdopen(filehandle, 'w')
+inventory_lock = open(inventorylock_path, "w+")
+os.chmod(inventorylock_path , stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
```

Comment:
I agree that this is the behavior it should have. @mthompson83 can clarify on what I mean. There is a non-zero processing time gap between your file creation call and the chmod call, which means that an attacker has the opportunity to act maliciously between those two lines.